llvm/lib/SYCLLowerIR/LowerESIMDVecArg.cpp

```diff
@@ -255,7 +255,9 @@ PreservedAnalyses ESIMDLowerVecArgPass::run(Module &M,
 
   SmallVector<Function *, 10> functions;
   for (auto &F : M) {
-    functions.push_back(&F);
+    // Skip functions that are used through function pointers
+    if (!F.hasAddressTaken())
```

Comment:
- This pass is not optional and is needed for correctness, I believe (see comments at the top of the file). So is this really safe to skip?
- We never tested function pointers support in ESIMD and never claimed they are supported, so I'm curious what is the ESIMD test with function pointers that fails?

Reply:
According to @pratikashar, skipping some function should be OK: "Just that VC may not be able to hold the value in GRF and it could lead to memory references."

All the tests in https://github.com/intel/llvm-test-suite/tree/intel/SYCL/ESIMD/fp_args_size were failing.

The simplest repro is this:
```cpp
ESIMD_NOINLINE void func(simd<int, SIZE> arg) {
}

using func_ptr = decltype(func);

SYCL_EXTERNAL SYCL_ESIMD_FUNCTION void init_ptr(func_ptr *&foo) {
  foo = &func;
}

SYCL_EXTERNAL SYCL_ESIMD_FUNCTION void use_ptr(func_ptr *foo) {
  simd<int, SIZE> A;
  foo(A);
}

SYCL_EXTERNAL SYCL_ESIMD_FUNCTION void esimd_kernel() {
  func_ptr* fp;
  init_ptr(fp);
  use_ptr(fp);
}
```

Reply:
Thanks. Please add this comment about safety to skip to the code.

Re function pointers - I see, this is 100% device side function pointer management, unlike the SYCL function pointers feature (see e.g. https://github.com/intel/llvm/blob/sycl/sycl/test/function-pointers/pass-fp-through-buffer.cpp).

Reply:
Done.